src/pudl/workspace/datastore.py

```diff
@@ -29,6 +30,7 @@
 # long as we stick to read-only keys.
 
 PUDL_YML = Path.home() / ".pudl.yml"
+ZenodoDOI = constr(regex=r"(10\.5072|10\.5281)/zenodo.([\d]+)")
```

Comment:
This defines the type, but doesn't actually do any validation. I use the `ZenodoDOI.validate()` method in the `ZenodoFetcher.__init__()` to check.